services/horizon/internal/app.go

```
@@ -357,28 +368,35 @@ func (a *App) UpdateStellarCoreInfo() {
 		return
 	}
 
-	core := &stellarcore.Client{
-		URL: a.config.StellarCoreURL,
+	var coreInfo *proto.InfoResponse
+	var err error
+	for i := 0; i < maxUpdateLedgerStateAttempts; i++ {
+		coreInfo, err = a.getCoreInfo()
+		if err == nil {
+			break
+		}
 	}
-
-	resp, err := core.Info(context.Background())
 	if err != nil {
-		log.Warnf("could not load stellar-core info: %s", err)
+		log.WithField("err", err.Error()).Warnf(
+			"failed to obtain stellar-core info after %v retries",
+			maxUpdateLedgerStateAttempts,
+		)
+		a.coreSettings.setNotSynced()
```

Should we assume "not synced" and unset it if this succeeds rather than assuming "synced" and setting if it fails?

---

if the request succeeds we will set the synced field based on the response structure of the response. this happens a few lines below when we call `	a.coreSettings.set(coreInfo)`.